src/browser/webapi/Node.zig

```diff
@@ -227,6 +227,81 @@ pub fn nodeType(self: *const Node) u8 {
     };
 }
 
+pub fn isEqualNode(self: *Node, other: *Node, page: *Page) bool {
+    // Make sure types match.
+    if (self.nodeType() != other.nodeType()) {
+        return false;
+    }
+
+    // TODO: Compare `localName` and prefix.
+    switch (self._type) {
+        .element => {
+            const e1 = self.as(Element);
+            const e2 = other.as(Element);
+
+            const e1_tag = e1.getTagNameSpec(&page.buf);
+            var buf: [Page.BUF_SIZE]u8 = undefined;
+            const e2_tag = e2.getTagNameSpec(&buf);
+
+            // Compare attributes.
+            var e1_iter = e1.attributeIterator();
+            var e2_iter = e2.attributeIterator();
+            var iter1_count: usize = 0;
+            var iter2_count: usize = 0;
+
+            while (e1_iter.next()) |attrib1| : (iter1_count += 1) {
+                const attrib2 = e2_iter.next() orelse return false;
+                iter2_count += 1;
+
+                if (attrib1._name.eql(attrib2._name) and attrib1._value.eql(attrib2._value)) {
+                    continue;
+                }
+
+                return false;
+            }
+
+            if (e1._namespace != e2._namespace or !std.mem.eql(u8, e1_tag, e2_tag) or iter1_count != iter2_count) {
+                return false;
+            }
+        },
+        .attribute => {
+            const a1 = self.as(Element.Attribute);
+            const a2 = other.as(Element.Attribute);
+            return std.mem.eql(u8, a1.getName(), a2.getName()) and std.mem.eql(u8, a1.getValue(), a2.getValue());
+        },
+        .cdata => {
+            const c1 = self.as(CData);
+            const c2 = other.as(CData);
+
+            return std.mem.eql(u8, c1.getData(), c2.getData());
+        },
+        else => @panic("NIY"),
+    }
+
+    // `Node.childNodes` allocate memory, this seems like a better option.
```

Comment:
```suggestion
```

Reply:
Yes, childrenIterator is meant for internal iteration within the Zig code, specifically to avoid the overhead of childNodes necessary to fulfill the webapi of that type.